src/Domain/Python/FunctionsOfTime.cpp

```diff
@@ -21,22 +23,99 @@ namespace domain::py_bindings {
 
 void bind_functions_of_time(py::module& m) {  // NOLINT
   domain::FunctionsOfTime::register_derived_with_charm();
-  py::class_<FunctionsOfTime::FunctionOfTime>(m, "FunctionOfTime")
+  py::class_<FunctionsOfTime::FunctionOfTime,
+             std::shared_ptr<FunctionsOfTime::FunctionOfTime>>(m,
+                                                               "FunctionOfTime")
       .def("time_bounds", &FunctionsOfTime::FunctionOfTime::time_bounds)
       .def("func", &FunctionsOfTime::FunctionOfTime::func, py::arg("t"))
       .def("func_and_deriv", &FunctionsOfTime::FunctionOfTime::func_and_deriv,
            py::arg("t"))
       .def("func_and_2_derivs",
            &FunctionsOfTime::FunctionOfTime::func_and_2_derivs, py::arg("t"));
+  m.def(
+      "serialize_functions_of_time",
+      [](const std::unordered_map<
+          std::string,
+          std::shared_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+             shared_deserialized_functions_of_time) {
+        std::unordered_map<
+            std::string,
+            std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
+            unique_deserialized_functions_of_time{};
+        if (shared_deserialized_functions_of_time.contains("Expansion")) {
```

Comment:
Just loop over all keys in the input map and call `get_clone`. No need to get specific names